Add all the periods

florelis opened the pull request:
Off-shot of @pressRtowin's #5279 

This adds periods at the end of all string resources for argument descriptions and error codes. It also replaces a bunch of semicolons by periods.

yao-msft commented:
Some of the strings are command/argument description. I thought we didn't include period on purpose?
![image](https://github.com/user-attachments/assets/0f97574e-5703-4ed3-9ab3-205b5238ed93)

Do we want to revisit the strings based on use cases?

florelis commented:
> Some of the strings are command/argument description. I thought we didn't include period on purpose? ![image](https://private-user-images.githubusercontent.com/50888816/420502197-0f97574e-5703-4ed3-9ab3-205b5238ed93.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDEzODIxNTMsIm5iZiI6MTc0MTM4MTg1MywicGF0aCI6Ii81MDg4ODgxNi80MjA1MDIxOTctMGY5NzU3NGUtNTcwMy00ZWQzLTlhYjMtMjA1YjUyMzhlZDkzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzA3VDIxMTA1M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUwYjhjZWFjOWM1NjRkNTA5NDg5YjE0M2M3ZjU5NTRiOGM4MmFjNDAyYjZhZjdmYzExNjAzZDEwZjVhN2QwZjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.F2axTKEaE1qNhw7z0IckXi3OiOiDzmaKyPFkF0GP9jo)
> 
> Do we want to revisit the strings based on use cases?

I did specifically change the argument descriptions and error messages. Currently most don't have periods, but some do. I just want to make it consistent in either direction. If what we want is for them to not have periods, that's cool by me.

I went to the side of yes periods because some strings had two sentences and it felt weird to only have the first one have a period in the first one. For example, `List packages even if their current version cannot be determined. Can only be used with the --upgrade-available argument`. Admittedly, all those seem to be my fault and could probably be reworded to be a single sentence 😅 

I also wondered if there was a standard for CLI tools and noticed that both `powershell -?` and `dotnet -?` use periods

denelon commented:
@pressRtowin I didn't see the other PR you made.

THANK YOU SO MUCH!

There are so many cases where we've failed to be consistent. It's been a challenge to prioritize cleaning up the mess. I really appreciate the time taken to review/improve the situation.

I was just talking with another PM this afternoon about the UX for the WinGet CLI. I'm going to point her at this PR as well just for the sake of keeping her informed.

I don't want to introduce any breaking changes, but the other area I've struggled with is in the naming of the commands. Most of them are verbs like "install" and "upgrade". Some of them are nouns like "settings" and "features". A couple of them could be ambiguous like "hash" and "pin".

Trenly commented:
I'd also like to chime in, especially with regards to this specific thought -

> As long as we're getting pedantic about grammar, there's also very inconsistent dropping of articles, even within the same sentence. "The path to the manifest of the package." elects to use every article, but "Override arguments to be passed on to the installer." drops the article for "arguments" ("the"), but doesn't for "installer".

In some cases, and the example here is a great one, the intent is truly that the object of the sentence is the plural `arguments` where `Override` is an adjective describing the type of arguments, and in these cases the form `Override arguments` is just as correct as `Override the arguments`. It is just a matter of whether what is being described is an action or an input

florelis commented:
Talked with @johnmcpms offline. For argument descriptions the main priority is to keep them short since the terminal has a limited space, even if it is not perfectly grammatical. I'm leaving argument descriptions with no periods as that's what most already had. For consistency, I removed periods from the few arguments that had it, and changed the few ones that were two "sentences" to be only one.

For the errors it makes sense to me to have them be more proper English, plus it aligns with other Windows error codes.

JohnMcPMS commented on src/AppInstallerCLIPackage/Shared/Strings/en-us/winget.resw:
```diff
@@ -3018,7 +3018,7 @@ Please specify one of them using the --source option to proceed.</value>
   </data>
   <data name="ConfigurationWarningSetViewTruncated" xml:space="preserve">
-    <value>Some of the data present in the configuration file was truncated for this output; inspect the file contents for the complete content.</value>
+    <value>Some of the data present in the configuration file was truncated for this output. Inspect the file contents for the complete content.</value>
```
```suggestion
    <value>Some of the data present in the configuration file was truncated for this output. Inspect the file for the complete content.</value>
```

JohnMcPMS commented on src/AppInstallerCLIPackage/Shared/Strings/en-us/winget.resw:
```diff
@@ -2315,46 +2315,46 @@ Please specify one of them using the --source option to proceed.</value>
   </data>
   <data name="APPINSTALLER_CLI_ERROR_ICU_CONVERSION_ERROR" xml:space="preserve">
-    <value>ICU conversion error</value>
+    <value>ICU conversion error.</value>
```
Unless this is a txt abbreviation for "I see you", I don't think that this is a sentence.  There are many of these error strings that are more noun-like.  If we are going for proper grammar, they probably shouldn't have periods, or they should be converted to sentences like:
```suggestion
    <value>An ICU conversion error occurred.</value>
```

denelon commented:
It's been a while for me (I remember the Microsoft Manual of Style), but it looks like the latest "writing" guidance for US English is https://learn.microsoft.com/en-us/style-guide/welcome/

florelis commented:
Marking this as a draft because clearly there are still a lot of strings to fix 😅

florelis commented on src/AppInstallerCLIPackage/Shared/Strings/en-us/winget.resw:
```diff
@@ -598,11 +598,11 @@ They can be configured through the settings file 'winget settings'.</value>
   </data>
   <data name="SourceOpenPredefinedFailedSuggestion" xml:space="preserve">
-    <value>Failed to open the predefined source; please report to winget maintainers.</value>
+    <value>Failed to open the predefined source. Please report to winget maintainers.</value>
```
I changed it to say "winget's maintainers" but the style guide [says](https://learn.microsoft.com/en-us/style-guide/punctuation/apostrophes)

> **Note** Don't use the possessive form of Microsoft trademarks and product, service, or feature names.

denelon replied:
I just laughed out loud and almost ended up wearing my coffee.

denelon commented:
My 2 cents: As long as we're consistent with "ourselves" in WinGet I think we're heading in the right direction. I'd like to be a great example for others to follow (assuming we're doing it the right way and honoring the guidance given). Given the problems we have with terminal width and how word wrapping is implemented, the main reason to omit trailing periods is to avoid an unwanted line feed.

denelon commented:
"WinGet" is the product name. There are some scenarios where the lower case makes sense in terms of one referencing the "winget" command. It depends on whether one is referring to the product or the executable. It's also valid to have a reference to the WinGet CLI (Command Line Interface). <- Yes I have a hard time using WinGet Command Line Interface (CLI) properly too.
